api/v1/api.go

```diff
@@ -1,808 +0,0 @@
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-// http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package v1
-
-import (
-	"encoding/json"
-	"errors"
-	"fmt"
-	"net/http"
-	"regexp"
-	"sort"
-	"sync"
-	"time"
-
-	"github.com/go-kit/log"
-	"github.com/go-kit/log/level"
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/common/model"
-	"github.com/prometheus/common/route"
-	"github.com/prometheus/common/version"
-
-	"github.com/prometheus/alertmanager/api/metrics"
-	"github.com/prometheus/alertmanager/cluster"
-	"github.com/prometheus/alertmanager/config"
-	"github.com/prometheus/alertmanager/dispatch"
-	"github.com/prometheus/alertmanager/pkg/labels"
-	"github.com/prometheus/alertmanager/provider"
-	"github.com/prometheus/alertmanager/silence"
-	"github.com/prometheus/alertmanager/silence/silencepb"
-	"github.com/prometheus/alertmanager/types"
-)
-
-var corsHeaders = map[string]string{
-	"Access-Control-Allow-Headers":  "Accept, Authorization, Content-Type, Origin",
-	"Access-Control-Allow-Methods":  "GET, POST, DELETE, OPTIONS",
-	"Access-Control-Allow-Origin":   "*",
-	"Access-Control-Expose-Headers": "Date",
-	"Cache-Control":                 "no-cache, no-store, must-revalidate",
-}
-
-// Alert is the API representation of an alert, which is a regular alert
-// annotated with silencing and inhibition info.
-type Alert struct {
-	*model.Alert
-	Status      types.AlertStatus `json:"status"`
-	Receivers   []string          `json:"receivers"`
-	Fingerprint string            `json:"fingerprint"`
-}
-
-// Enables cross-site script calls.
-func setCORS(w http.ResponseWriter) {
-	for h, v := range corsHeaders {
-		w.Header().Set(h, v)
-	}
-}
-
-// API provides registration of handlers for API routes.
-type API struct {
-	alerts   provider.Alerts
-	silences *silence.Silences
-	config   *config.Config
-	route    *dispatch.Route
-	uptime   time.Time
-	peer     cluster.ClusterPeer
-	logger   log.Logger
-	m        *metrics.Alerts
-
-	getAlertStatus getAlertStatusFn
-
-	mtx sync.RWMutex
-}
-
-type getAlertStatusFn func(model.Fingerprint) types.AlertStatus
-
-// New returns a new API.
-func New(
-	alerts provider.Alerts,
-	silences *silence.Silences,
-	sf getAlertStatusFn,
-	peer cluster.ClusterPeer,
-	l log.Logger,
-	r prometheus.Registerer,
-) *API {
-	if l == nil {
-		l = log.NewNopLogger()
-	}
-
-	return &API{
-		alerts:         alerts,
-		silences:       silences,
-		getAlertStatus: sf,
-		uptime:         time.Now(),
-		peer:           peer,
-		logger:         l,
-		m:              metrics.NewAlerts("v1", r),
```

Comment:
Technically, we can remove this distinction now but it would be a rather critical breaking change - I would rather not risk it.

Reply:
nvm, I made it permament.